### PR TITLE
refactor: replace v8Util.createIDWeakMap() with native WeakRef in tests

### PR DESCRIPTION
**Pull request by miniak**
#### Checklist

- [ ] PR description included and stakeholders cc'd
- [ ] `npm test` passes
- [ ] tests are [changed or added](https://github.com/electron/electron/blob/master/docs/development/testing.md)
- [ ] relevant documentation is changed or added
- [ ] PR title follows semantic [commit guidelines](https://github.com/electron/electron/blob/master/docs/development/pull-requests.md#commit-message-guidelines)
- [ ] [PR release notes](https://github.com/electron/clerk/blob/master/README.md) describe the change in a way relevant to app developers, and are [capitalized, punctuated, and past tense](https://github.com/electron/clerk/blob/master/README.md#examples).
- [ ] This is **NOT A BREAKING CHANGE**. Breaking changes may not be merged to master until 11-x-y is branched.

#### Release Notes

Notes:

### CODE REVIEW

File: typings/internal-ambient.d.ts [hunk 1]
```diff
@@ -215,3 +215,71 @@ interface ResizeObserverEntry {
...
+  readonly prototype: FinalizationRegistry;
+
+  /**
+   * Creates a finalization registry with an associated cleanup callback
+   * @param cleanupCallback The callback to call after an object in the registry has been reclaimed.
+   */
+  new(cleanupCallback: (heldValue: any) => void): FinalizationRegistry;
+}
+
+declare var FinalizationRegistry: FinalizationRegistryConstructor;
```
Comment (nornagon) [inline typings/internal-ambient.d.ts:285]: Is this copy/pasted from somewhere? Perhaps note where?

Comment (miniak) [reply typings/internal-ambient.d.ts:285]: it's from https://github.com/microsoft/TypeScript/pull/38232

Comment (miniak) [reply typings/internal-ambient.d.ts:285]: @nornagon do you know why am I getting this linter error even though `WeakRef` is declared?
```
/home/builduser/project/src/electron/spec-main/api-browser-window-spec.ts
  74:23  error  'WeakRef' is not defined  no-undef

/home/builduser/project/src/electron/spec-main/api-menu-spec.ts
  845:23  error  'WeakRef' is not defined  no-undef
```

Comment (nornagon) [reply typings/internal-ambient.d.ts:285]: No, I'm not sure why that would be. Seems like `declare var WeakRef` on L239 should fix that?

**Comment by nornagon** [comment]
I landed some of this in https://github.com/electron/electron/pull/24116

**Pull request closed without merging**